service/raft/raft_group0.cc

```diff
@@ -1051,12 +1053,17 @@ static future<> check_remote_group0_upgrade_state_dry_run(
     static constexpr auto max_concurrency = 10;
 
     auto retry_period = std::chrono::seconds{1};
-    while (true) {
+    for(;;
```

Comment:
Would you say that splitting `for(;;` to a separate line makes this thing more readable?

Reply:
@tgrabiec what would you say?

Reply:
Splitting to the next line doesn't safe much characters in this case, I'd prefer it to be on a single line.

We could make the line shorter by moving retry_period update to `next_retry_period()`:
```
for (;; co_await sleep_abortable(next_retry_period(), as))) {
```

Reply:
sleep_with_backoff() approach works too.

Reply:
I added sleep_with_exponential_backoff class.